packages/monaco/src/browser/textmate/monaco-textmate-frontend-bindings.ts

```diff
@@ -25,8 +25,29 @@ import { LanguageGrammarDefinitionContribution } from './textmate-contribution';
 import { MonacoTextmateService, OnigasmPromise } from './monaco-textmate-service';
 import { loadWASM } from 'onigasm';
 
+export function loadOnigasm(): Promise<ArrayBuffer> {
+    return new Promise((resolve, reject) => {
+        const onigasmPath = require('onigasm/lib/onigasm.wasm'); // webpack doing its magic here
+        const request = new XMLHttpRequest();
+
+        request.onreadystatechange = function () {
+            if (this.readyState === XMLHttpRequest.DONE) {
+                if (this.status === 200) {
+                    resolve(this.response);
+                } else {
+                    reject(new Error('Could not fetch onigasm'));
+                }
+            }
+        };
```

Comment:
would the code be easier to read if we use `xhr-request` (https://www.npmjs.com/package/xhr-request#example)? on the other hand, with `new XMLHttpRequest()` we don't need to add a new dependency :)